flyteplugins/go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -746,20 +746,20 @@ func MergeBasePodSpecOntoTemplate(templatePodSpec *v1.PodSpec, basePodSpec *v1.P
 					return nil, err
 				}
 			}
-		}
-
-		// Check for any name matching template containers
-		for _, templateContainer := range templatePodSpec.Containers {
-			if templateContainer.Name != container.Name {
-				continue
-			}
+		} else {
```

Comment:
This change seems okay to me. Can we have the same behavior for init containers to keep it consistent?